cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -100,6 +103,19 @@ private void checkServiceResources(Set<String> problemKeys) {
     }
   }
 
+  private void checkBindAddress() {
+    // check if app.bind.address is loopback address
+    String appBindAddress = cConf.get(Constants.AppFabric.SERVER_ADDRESS);
+    try {
+      if (InetAddress.getByName(appBindAddress).isLoopbackAddress()) {
+        LOG.warn("{} is set to {}. This is invalid in any multinode Hadoop cluster.",
```

Comment:
still think "invalid" doesn't cut the point across. It should say that the service may not be discoverable on a multinode Hadoop cluster.